src/hardware/serialport/serialport.cpp

```diff
@@ -1211,7 +1211,7 @@ bool CSerial::Getchar(uint8_t *data, uint8_t *lsr, bool wait_dsr, uint32_t timeo
 		}
 	}
 	// wait for a byte to arrive
-	while ((!((*lsr = Read_LSR()) & LSR_RX_DATA_READY_MASK)) &&
+	while (!((*lsr = static_cast<uint8_t>(Read_LSR())) & LSR_RX_DATA_READY_MASK) &&
```

Comment:
Uh, this looks worrying. I hope loss of precision is acceptable in these cases.

It's definitely better to have it happen explicitly though!